docs/LogGuide.md

```diff
@@ -328,65 +428,125 @@ and does not consistently show up first.
 
-Examples:
+#### Structure
+
+```log
+Network Log Line Structure:
+01|[timestamp]|[id]|[name]
+
+ACT Log Line Structure:
+[timestamp] 01:Changed Zone to [Name].
+```
+
+#### Regexes
 
 ```log
-01:Changed Zone to The Lavender Beds.
-01:Changed Zone to The Unending Coil Of Bahamut (Ultimate).
+Network Log Line Regex:
+^(?<type>(?:01))\|(?<timestamp>(?:[^|]*))\|(?<id>(?:[^|]*))\|(?<name>(?:[^|]*))\|
+
+ACT Log Line Regex:
+(?<timestamp>(?:^.{14})) 01:Changed Zone to (?<name>(?:.*?))\.
+```
+
+#### Examples
+
+```log
+Network Log Line Examples:
+01|2021-04-26T14:13:17.9930000-04:00|326|Kugane Ohashi|b9f401c0aa0b8bc454b239b201abc1b8
+01|2021-04-26T14:22:04.5490000-04:00|31f|Alphascape (V2.0)|8299b97fa36500118fc3a174ed208fe4
+
+ACT Log Line Examples:
+[14:13:17.993] 01:Changed Zone to Kugane Ohashi.
+[14:22:04.549] 01:Changed Zone to Alphascape (V2.0).
 ```
 
+<!-- AUTO-GENERATED-CONTENT:END -->
+
 ### 02: ChangePrimaryPlayer
 
 This redundant message follows every [ChangeZone](#01-changezone) message to indicate the name of the player.
 
-Structure:
-`02:Changed primary player to [Player Name].`
+<!-- AUTO-GENERATED-CONTENT:START (logLines:type=ChangedPlayer&lang=en) -->
+
+#### Structure
+
+```log
+Network Log Line Structure:
+02|[timestamp]|[id]|[name]
 
-Examples
+ACT Log Line Structure:
+[timestamp] 02:Changed primary player to [name].
+```
+
+#### Examples
 
 ```log
-02:Changed primary player to Potato Chippy.
-02:Changed primary player to Tini Poutini.
+Network Log Line Examples:
+02|2021-04-26T14:11:31.0200000-04:00|10ff0001|Tini Poutini|5b0a5800460045f29db38676e0c3f79a
+02|2021-04-26T14:13:17.9930000-04:00|10ff0002|Potato Chippy|34b657d75218545f5a49970cce218ce6
+
+ACT Log Line Examples:
+[14:11:31.20] 02:Changed primary player to Tini Poutini.
+[14:13:17.993] 02:Changed primary player to Potato Chippy.
 ```
 
+<!-- AUTO-GENERATED-CONTENT:END -->
+
 ### 03: AddCombatant
 
 This message is sent when a new object is added to the scene or
 becomes close enough to the player that they can view its actions.
 
-Structure:
-`03:[ObjectId]:Added new combatant [Combatant Name].  Job: [Job-ID] Level: [Level-Value] Max HP: [Max-HP-Value] Max MP: [Max-MP-Value] Pos: ([X-Pos],[Z-Pos],[Y-Pos]).`
+<!-- AUTO-GENERATED-CONTENT:START (logLines:type=AddedCombatant&lang=en) -->
 
-Examples:
+#### Structure
 
 ```log
-03:40123456:Added new combatant Pagos Deepeye.  Job: N/A Level: 70 Max HP: 348652 Max MP: 12000 Pos: (-720.9337,90.80706,-679.6056).
-03:10987654:Added new combatant Tater Tot (Jenova).  Job: 28 Level: 70 Max HP: 39835 Max MP: 16461 Pos: (-143.9604,168.5795,-4.999999).
+Network Log Line Structure:
+03|[timestamp]|[id]|[name]|[job]|[level]|[ownerId]|[worldId]|[world]|[npcNameId]|[npcBaseId]|[currentHp]|[hp]|[currentMp]|[mp]|[?]|[?]|[x]|[y]|[z]|[heading]
+
+ACT Log Line Structure:
+[timestamp] 03:[ID]:Added new combatant [name]([world]).  Job: NONE Level: [level] Max HP: [hp] Max MP: [mp] Pos: ([x],[y],[z]) ([npcNameId][npcBaseId]).
```

Comment:
This can't be fixed in the same way as the damage issue unfortunately, since having a blank job is a valid case that defaults to NONE for e.g. enemy combatants. I think this one will have to be a special case in the generator instead?

Reply:
Did you want to solve this before I merge this?

Reply:
Solving this would basically be something like this in the transform function:

```ts
if (type === 'AddedCombatant')
  structureLog = structureLog.replace(/Job: NONE/, 'Job: [job]')
```

Which is pretty ugly. If you think that's an okay fix then sure, I can add it.

Reply:
I don't know.  I think having the log structure be clearer is better than having a clean script, so go for it?  Agreed that it's pretty ugly but I can't see another way to do it.